frontend/packages/shared/src/components/GiteaHeader/VersionControlButtons/components/ShareChangesPopover/CommitAndPushContent/FileChangesInfoModal/FileChangesInfoModal.tsx

```diff
@@ -81,17 +89,53 @@ export const FileChangesInfoModal = ({
             </Table.Row>
           </Table.Head>
           <Table.Body>
-            {fileChanges.map((fileChange) => {
-              return (
-                <Table.Row key={fileChange.filePath}>
-                  <Table.Cell>{renderFilePath(fileChange.filePath)}</Table.Cell>
-                  <Table.Cell>{renderFileStatusTag(fileChange.fileStatus)}</Table.Cell>
-                </Table.Row>
-              );
-            })}
+            {fileChanges.map((fileChange) => (
+              <FileChangeTableRow
+                key={fileChange.filePath}
+                fileChange={fileChange}
+                diff={gitDiffIncludesFile(fileChange.filePath) && repoDiff[fileChange.filePath]}
+                repoDiffStatus={repoDiffStatus}
+              />
+            ))}
           </Table.Body>
         </Table>
+        {renderRepoDiffStatus()}
       </div>
     </StudioModal>
   );
 };
+
+interface FileChangeTableRowProps {
+  fileChange: RepoContentStatus;
+  diff?: string; // Might be null for deleted files
+  repoDiffStatus: 'success' | 'error' | 'pending';
+}
+
+const FileChangeTableRow = ({ fileChange, diff, repoDiffStatus }: FileChangeTableRowProps) => {
+  const { t } = useTranslation();
+
+  const renderFileStatusTag = (): React.ReactElement => {
+    return (
+      <Tag size='small' color={fileStatusToTagColorMapping[fileChange.fileStatus]}>
+        {t(`sync_header.show_changes_modal.file_status_${fileChange.fileStatus}`)}
+      </Tag>
+    );
+  };
```

Comment:
Dette trenger vel ikke å være en funksjon i det hele tatt?
```suggestion
  const fileStatusTag: React.ReactElement => (
    <Tag size='small' color={fileStatusToTagColorMapping[fileChange.fileStatus]}>
      {t(`sync_header.show_changes_modal.file_status_${fileChange.fileStatus}`)}
    </Tag>
  );
```

`fileStatus` brukes for øvrig mange ganger her, så du kan godt vurdere å destrukturere `fileChange`;
`const {fileStatus, filePath} = fileChange`

Reply:
Her har jeg blingset og glemt å bytte ut `=>` med `=`. Det er nok derfor testen feiler, i tillegg til at jeg har endret navn på variabelen.